types/vote_test.go

```diff
@@ -130,12 +130,17 @@ func TestVoteSignBytesTestVectors(t *testing.T) {
 		},
 		// containing vote extension
 		5: {
-			"test_chain_id", &Vote{Height: 1, Round: 1, VoteExtension: VoteExtension{
-				AppDataToSign:             []byte("signed"),
-				AppDataSelfAuthenticating: []byte("auth"),
-			}},
+			"test_chain_id", &Vote{
+				Type:           0,
+				Height:         1,
+				Round:          1,
+				BlockID:        BlockID{},
+				Timestamp:      time.Time{},
+				ValidatorIndex: 0,
+				Extension:      []byte("signed"),
+			},
 			[]byte{
-				0x38,                                   // length
+				0x2e,                                   // length
```

Comment:
This is pre-existing, but I find it deeply weird that we have hardcoded golden wire-format messages in this test. It seems like that makes a change-detector rather than a real test, since if the real marshaling code emits something different, the signatures won't verify correctly in the real code but will still pass these hardwired tests.

Not something we need to resolve right now, but I'm not sure what we imagine to be testing here.

Reply:
Yeah this does ultimately seem like a change detector. But perhaps this is something that Buf should be detecting for us instead?

I found this nifty [protobuf decoder](https://protobuf-decoder.netlify.app/) that I used with the hex values from our test structs to ensure I understood what I was changing before I changed it. I'm sure there's a CLI-based tool for this somewhere (and if there isn't there should be).

Reply:
I was thinking more in the other direction: We can populate the (generated) structs with the data we expect to verify, marshal them, and then verify that the results plumb through correctly. But that would work too. 

I'm slightly biased toward making the test cases easier for a human to read, which would militate against hex strings, but I'm sure this is something we can revisit.

Reply:
@creachadair, I had the same thought when I came across these tests in January. Probably something we could improve on if we all agree it'd be more readable and there are no drawbacks we may be missing here

Reply:
In my understanding these tests effectively do live as 'change detectors'.
The goal of these tests is to ensure that our different types continue to marshal to the exact same set of bytes across different versions of the code. The purpose of that being to ensure that our signed/hashed data isn't changing as we make updates to Tendermint.